pkg/controller/queuejobresources/genericresource/genericresource.go

```diff
@@ -450,3 +451,59 @@ func getContainerResources(container v1.Container, replicas float64) *clustersta
 	req.GPU = req.GPU * int64(replicas)
 	return req
 }
+
+//return value is a string for flexibility of expression
+func (gr *GenericResources) GetGenericItemKindStatus(aw *arbv1.AppWrapperGenericResource, namespace string) (completed string) {
+	dd := gr.clients.Discovery()
+	apigroups, err := restmapper.GetAPIGroupResources(dd)
+	if err != nil {
+		klog.Errorf("Error getting API resources, err=%#v", err)
+	}
+	restmapper := restmapper.NewDiscoveryRESTMapper(apigroups)
+	_, gvk, err := unstructured.UnstructuredJSONScheme.Decode(aw.GenericTemplate.Raw, nil, nil)
+	if err != nil {
+		klog.Errorf("Decoding error, please check your CR! Aborting handling the resource creation, err:  `%v`", err)
+	}
+	//service and podgroup has no conditions return ALWAYS true
+	//TODO: make it configurable?
+	if gvk.GroupKind().Kind == "Service" || gvk.GroupKind().Kind == "PodGroup" {
+		return "ignore"
+	}
+
```

Comment:
This can be removed if generic item does not have the new field `completionstatus` correct?

Reply:
Yes, according to the protocol the user is expected to provide us with conditions for the correct object. the user hsuld not provide `completionrequired` property for service or podgroup. I will remove this code.

Reply:
I don't think so, service or podgroup does not have completion status. from our initial design the idea was to hold AW in RunningHoldCompletion if any item other than podgroup or service is not complete.
![Screen Shot 2022-08-01 at 6 15 15 PM](https://user-images.githubusercontent.com/5898557/182254821-5caf5250-f5c3-463b-8758-9aeb57641f5c.png)

Reply:
Ok, I see the challenge here.   As part of the AW pod phases (pending/running/completed...) are tracked.  If there is a check of the pods when setting state, this would be sufficient.  Iterating over each generic item that is not tracked would not be necessary.  only when all tracked items indicate `completionstatus` the `state` should be changed (of which is being done).  The change here is that the decision to set state to `RunningHoldCompletion` or `Complete` is when all pods associated to AW are in Failed/Completed phase states.  These counts are already tracked in the AW and can be referenced from the `.status` field.  Please let me know your thoughts.  Thanks!

Reply:
please refer to this comment https://github.com/IBM/multi-cluster-app-dispatcher/pull/192#discussion_r936569915